modin/core/dataframe/pandas/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,467 @@
+            "b": DTypeKind.BOOL,
+        }
+        kind = _np_kinds.get(dtype.kind, None)
+        if kind is None:
+            raise NotImplementedError(
+                f"Data type {dtype} not supported by the dataframe exchange protocol"
+            )
+        return (
+            kind,
+            dtype.itemsize * 8,
+            pandas_dtype_to_arrow_c(dtype),
+            dtype.byteorder,
+        )
+
+    @property
+    def describe_categorical(self) -> Dict[str, Any]:
+        if not self.dtype[0] == DTypeKind.CATEGORICAL:
+            raise RuntimeError(
+                "`describe_categorical only works on a column with "
+                + "categorical dtype!"
+            )
+
+        cat_dtype = self._col.dtypes[0]
+        ordered = cat_dtype.ordered
+        is_dictionary = True
+        # NOTE: this shows the children approach is better, transforming
+        # `categories` to a "mapping" dict is inefficient
+        # codes = self._col.values.codes  # ndarray, length `self.size`
+        # categories.values is ndarray of length n_categories
+        categories = cat_dtype.categories
+        mapping = {ix: val for ix, val in enumerate(categories)}
+        return {
+            "is_ordered": ordered,
+            "is_dictionary": is_dictionary,
+            "mapping": mapping,
+        }
+
+    @property
+    def describe_null(self) -> Tuple[int, Any]:
+        _k = DTypeKind
+        kind = self.dtype[0]
+        value = None
+        if kind == _k.FLOAT:
+            null = 1  # np.nan
+        elif kind == _k.DATETIME:
+            null = 1  # np.datetime64('NaT')
+        elif kind in (_k.INT, _k.UINT, _k.BOOL):
+            # TODO: check if extension dtypes are used once support for them is
+            #       implemented in this protocol code
+            null = 0  # integer and boolean dtypes are non-nullable
+        elif kind == _k.CATEGORICAL:
+            # Null values for categoricals are stored as `-1` sentinel values
+            # in the category date (e.g., `col.values.codes` is int8 np.ndarray)
+            null = 2
+            value = -1
+        elif kind == _k.STRING:
+            null = 4
+            value = (
+                0  # follow Arrow in using 1 as valid value and 0 for missing/null value
+            )
+        else:
+            raise NotImplementedError(f"Data type {kind} not yet supported")
+
+        return null, value
+
+    _null_count_cache = None
+
+    # TODO: since python 3.9:
+    # @cached_property
+    @property
+    def null_count(self) -> int:
+        if self._null_count_cache is not None:
+            return self._null_count_cache
+
+        def map_func(df):
+            return df.isna()
+
+        def reduce_func(df):
+            return pandas.DataFrame(df.sum())
+
+        intermediate_df = self._col.tree_reduce(0, map_func, reduce_func)
+        intermediate_df.index = pandas.RangeIndex(1)
+        intermediate_df.columns = pandas.RangeIndex(1)
```

Comment:
really hacky but can work for now I guess, but let's put a descriptive comment for a better understanding of this segment if/when the hack would stop working someday

Reply:
Added a comment